pkg/noobaaaccount/noobaaaccount.go

```diff
@@ -707,6 +763,152 @@ func GenerateNonCrdAccountKeys(name string) error {
+	accountInfo, err := sysClient.NBClient.ReadAccountAPI(nb.ReadAccountParams{
+		Email: name,
+	})
+	if err != nil {
+		return err
+	}
+
+	accessKeys = accountInfo.AccessKeys[0]
+
+	secret.StringData = map[string]string{}
+	secret.StringData["AWS_ACCESS_KEY_ID"] = accessKeys.AccessKey
+	secret.StringData["AWS_SECRET_ACCESS_KEY"] = accessKeys.SecretKey
+
+	// If we will not be able to update the secret we will print the credentials as they already been changed by the RPC
+	if !util.KubeUpdate(secret) {
+		log.Printf(`❌  Please verify the updated credentials for account %s:`, name)
+		fmt.Printf("\nAWS_ACCESS_KEY_ID     : %s\n", accessKeys.AccessKey)
+		fmt.Printf("AWS_SECRET_ACCESS_KEY : %s\n\n", accessKeys.SecretKey)
+		log.Fatalf(`❌  Failed to update the secret %s with the new accessKeys`, secret.Name)
+	}
+
+	log.Printf("✅ Successfully updated s3 credentials for the account %q", name)
+	return nil
+}
+
+// UpdateNonCrdAccountKeys update noobaa account (none CRD based) S3 keys
+func UpdateNonCrdAccountKeys(name string, accessKeys nb.S3AccessKeys) error {
+	log := util.Logger()
+
+	sysClient, err := system.Connect(true)
+	if err != nil {
+		return err
+	}
+
+	err = sysClient.NBClient.UpdateAccountKeysAPI(nb.UpdateAccountKeysParams{
+		Email:      name,
+		AccessKeys: accessKeys,
+	})
+	if err != nil {
+		if nbErr, ok := err.(*nb.RPCError); ok && nbErr.RPCCode == "NO_SUCH_ACCOUNT" {
+			log.Fatalf(`❌  Could not find the account: %s, will not update keys.`, name)
+		}
+		return err
+	}
+
+	// UpdateAccountKeysAPI have no reply so we need to read the account in order to get the updated credentials
+	accountInfo, err := sysClient.NBClient.ReadAccountAPI(nb.ReadAccountParams{
+		Email: name,
+	})
+	if err != nil {
+		log.Fatalf(`❌  Could not read account: %s, keys were allready updated, please read the account to get the keys`, name)
+	}
+
+	accessKeys = accountInfo.AccessKeys[0]
+
+	log.Printf("✅ Successfully update s3 credentials for the account %q", name)
+	fmt.Printf("\nAWS_ACCESS_KEY_ID     : %s\n", accessKeys.AccessKey)
+	fmt.Printf("AWS_SECRET_ACCESS_KEY : %s\n\n", accessKeys.SecretKey)
+
+	return nil
+}
+
+// ValidateAccessKeys checks for complexity of credentials
+func ValidateAccessKeys(accessKeys nb.S3AccessKeys) {
+	log := util.Logger()
+
+	// Checking complexity for access_key
+	if !ValidateComplexity(accessKeys.AccessKey, 20, true, false, true, false) {
+		log.Fatalf(`❌ Account access key length must be 20, and must contain uppercase and numbers`)
+	}
+
+	// Checking complexity for secret_key
+	if !ValidateComplexity(accessKeys.SecretKey, 40, true, true, true, true) {
+		log.Fatalf(`❌ Account secret length must be 40, and must contain uppercase, lowercase, numbers and symbols`)
+	}
+}
+
+// ValidateComplexity checks for complexity
+func ValidateComplexity(str string, length int, upper bool, lower bool, numbers bool, symbols bool) bool {
```

Comment:
To avoid the long list of variables in the function signature maybe use a struct.
Since you are passing the list of boolean, which is hard-coded, I think it would be better to use defaults like you have in core:
https://github.com/noobaa/noobaa-core/blob/ff0007ad2512b0f67efa80112ceb2fcea0f3a63a/src/util/string_utils.js#L226-L242

Reply:
Sure, Thanks.